---
layout: page
title: Projects
---

<br>
<br>
<br>


<!--
<strong> <font size="4"> Fast and Robust Inference </font> </strong>
<p> The exponential growth in size and dimensionality of the modern data has increased the complexity of the current machine learning models; the size of machine learning models is also growing exponentially. Bulkier machine learning models are harder to train, slower to test, and less relaible agianst adversarial attacks. I use techniques from large-scale optimization and streaming literature to develop fast and robust inference algorithms.
</p>
-->


<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/landscape.png" alt="Alt Text" width="500">
            <figcaption class="caption">A rugged fitness landscape is shown. Figure adapted from the <a href="https://en.wikipedia.org/wiki/Sequence_space_(evolution)">Wiki page</a>.</figcaption>
    </div>


    <div class="toright">
        <p> <b>CRISPRLand</b> is a platform to analyze the machine learning algorithms trained on the samples from the fitness landscape of DNA repair after CAS9-induced breaks in the Walsh-Hadamard spectral domain. CRISPRLand provides an exponential speed-up in inferring the DNA repair landscape and redution in sample complexity (CRISPR experiments).
        </p>
           
                
                [C1] <b>A. Aghazadeh</b>, Orhan Ocal, and Kannan Ramchandran, <a href="https://www.nature.com/articles/s41587-019-0203-2">CRISPRLand: Interpretable Large-Scale Inference of DNA Repair Landscape Based on a Spectral Approach</a>, <em> International Society for Computational Biology (ISMB)</em>, July 2020. <br><br>
                
                [C2] <b>A. Aghazadeh</b>, Orhan Ocal, and Kannan Ramchandran, <a href="https://www.nature.com/articles/s41587-019-0203-2">CRISPRLand: Interpretable Large-Scale Inference of DNA Repair Landscape Based on a Spectral Approach</a>, <em> Bioinformatics</em>, July 2020. <br><br>
        
        
        <a href="https://github.com/UCBASiCS/CRISPRLand">Code on Github</a> <br>

        
    </div>
</div>




<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/crispr.png" alt="Alt Text" width="500">
            <figcaption class="caption">DNA repair after a double-strand break. Figure adapted from the <a href="https://scopeblog.stanford.edu/2019/07/29/crispr-algorithm-predicts-how-well-gene-editing-will-work/">Stanford SCOPE</a> covering our paper, titled "CRISPR algorithm predicts how well gene editing will work".</figcaption>
    </div>


    <div class="toright">
        <p> <b>SPROUT</b> is a machine learning algorithm based on random forests to predict the DNA repair outcome after CAS9-induced double-strand breaks in T Cells. Our analysis of DNA repair outcome using SPROUT reveals long insertions (> 20 nucleotides) from spacially close locations on the genome.
        </p>
        
                
                [C3] R. Leenay*, <b>A. Aghazadeh</b>*, J. Hiatt*, D. Tse, T. L. Roth, R. Apathy, E. Shifrut, J. F. Hulquist, N. Krogen, Z. Wu, G. Cirolia, H. Canaj, M. Leonetti, A. Marson, A. May, and J. Zou, <a href="https://www.nature.com/articles/s41587-019-0203-2">Large dataset enables prediction of repair after CRISPR–Cas9 editing in primary T cells</a>, <em>Nature Biotechnology</em>, July 2019. <br><br>
        
        
        <a href="https://zou-group.github.io/SPROUT">SPROUT interactive software</a> <br>
        <a href="https://github.com/amirmohan/SPROUT">Code on Github</a> <br>
        <a href="https://figshare.com/projects/Systematic_characterization_of_genome_editing_in_primary_T_cells_reveals_proximal_genomic_insertions_and_enables_machine_learning_prediction_of_CRISPR-Cas9_DNA_repair_outcomes/37166">Data on figshare</a> <br> <br>
        
        News Coverage <br>
        <a href="https://scopeblog.stanford.edu/2019/07/29/crispr-algorithm-predicts-how-well-gene-editing-will-work/">Stanford University</a> <br>
        
    </div>
</div>







<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/sketch.png" alt="Alt Text" width="500">
            <figcaption class="caption">Count Sketch is a data structure from streaming literature. Photo adapted from   <a href="https://blog.acolyer.org/2016/07/21/time-adaptive-sketches-ada-sketches-for-summarizing-data-streams/"><em>the morning paper</em></a> a blog post on time-adaptive sketches for summarizing data streams.</figcaption>
    </div>


    <div class="toright">
        <p> <b>MISSION</b> is a first-order and <b>BEAR</b> is a second-order optimization algorithm for  <em>sublinear-memory</em> <b>feature selection</b> in massively high-dimensional datasets. The key idea behind these algorithms is to store and iteratively update the model parameters in an efficient data structure from streaming called <em>Count Sketch</em>.
        </p>
        
                
        [S1] <b>A. Aghazadeh</b>*, R. Spring*, D. LeJeune, G. Dasarathy, A. Shrivastava and R. G. Baraniuk, <a href="https://icml.cc/Conferences/2018/ScheduleMultitrack?event=2916">MISSION: Ultra Large-Scale Feature Selection using Count Sketches</a>, <em>Proc. Intl. Conf. Machin. Learning (ICML)</em>, July 2018. <br>
        
        
        [S2] <b>A. Aghazadeh</b>, V. Gupta, A. Dewasee, O. Koyluoglu, and K. Ramchandran, BEAR: Sketching BFGS Algorithm for Feature Selection with Sublinear Memory, <em>Submitted</em>. <br> <br>
        
        <a href="https://github.com/rdspring1/MISSION">MISSION Code on Github</a> <br>
        <a href="https://github.com/rdspring1/MISSION">BEAR Code on Github</a>
        
    </div>
</div>







<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/infinity.png" alt="Alt Text" width="500">
            <figcaption class="caption">Schematic of p-norm balls.</figcaption>
    </div>


    <div class="toright">
        <p> <b>RHash</b> is a fast algorithm to learn <b>robust data-dependent hash functions</b>. It minimizes the worst-case L-infinity norm distortion instead of the more conventional average ditortion to learn hash functions.

        </p>
        
        
        [H1] <b>A. Aghazadeh</b>, A. S. Lan, A. Shrivastava and R. G. Baraniuk, <a href="https://www.ijcai.org/Proceedings/2017/0192.pdf">RHash: Robust Hashing via L∞-norm Distortion</a>, <em>Proc. Intl. Joint Conf. Artif. Intell. (IJCAI)</em>, Aug. 2017. <br>
        
        
    </div>
</div>



<!--
<strong><font size="4">Data, Computation, and Sensing</font></strong>
<p> I develop methods for computational sensing by blending the mathematics of computational models, the physical properties of the sensing medium, and the structure of measurement data. I have designed a novel molecular sensing platform for universal microbial diagnostics, which infuses the principles of DNA probe design with the sampling techniques inspired by compressive sensing. A key enbaler of our design is the growing genomic data sequenced from microbial organims everyday. I have also designed data-driven detection strategies for structural health monitoring.
      </p>
-->

<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/UMD.pdf" alt="Alt Text" width="500">
            <figcaption class="caption">Universal Microbial Diagnostic (UMD) platform shows the steps to identify the composition of a microbial sample using randomized DNA probes. Photo is adapted from our <em>Science Advances</em> paper [M1].</figcaption>
    </div>

    <div class="toright">
        <p> <b>Universal Microbial Diagnostics</b> (UMD) is a new sensing paradigm that we have develoepd at Rice University to detect microbial agents (e.g., superbugs and viruses) using a small number of target-agnostic randomized DNA probes. The inference algorithm solves a compressive sensing problem to identify the composition of the microbial samples.
        </p>
    
        
        [M1] <b>A. Aghazadeh</b>*, A. Y. Lin*, M. A. Sheikh*, R. A. Drezek, R. G. Baraniuk, et al., <a href="https://advances.sciencemag.org/content/2/9/e1600025?utm_source=TrendMD&utm_medium=cpc&utm_campaign=TrendMD_1">Universal microbial diagnostics using random DNA probes</a>, <em>Science Advances</em> 2, e1600025 (2016). <br>
        
        </p>
        
        <a href="https://patents.google.com/patent/US20180355411A1/en">US Patent</a> <br>
        <a href="https://www.ece.rice.edu/news/21704-REA-Awards">Hershel M. Rich Invention Award</a> <br><br>
    
        
        News Coverage <br>
        <a href="http://www.houstonchronicle.com/local/prognosis/article/New-probe-designed-by-Rice-and-Baylor-team-could-9368593.php">Houston Chronicle</a> <br>
        
        <a href="https://www.genomeweb.com/molecular-diagnostics/universal-microbial-diagnostics-approach-using-random-dna-probes-promises#.XqXD3y-z10s">GenomeWeb</a> <br>
        
        <a href="https://www.pharmaceutical-journal.com/news-and-analysis/research-briefing/scientists-create-universal-microbial-screening-method/20201883.article">The Pharmaceutical Journal</a> <br>

        <a href="https://www.biocentury.com/article/279557/the-first-catch-all-diagnostic-for-infections">BioCentury Innovations</a> <br>
        
        
        <a href="https://www.labroots.com/trending/microbiology/4207/technology-easily-identifies-bacterial-pathogens">Labroots</a> <br>
        
        
    </div>
</div>


<hr>



<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="canter" src="../assets/images/insense.pdf" alt="Alt Text" width="300">
            <figcaption class="caption">Universal Microbial Diagnostic (UMD) platform shows the steps to identify the composition of a microbial sample using randomized DNA probes. Photo is adapted from our <em>Science Advances</em> paper [M1].</figcaption>
    </div>

    <div class="toright">
        <p> <b>Insense</b> is an optimization algorithms to select optimal linear measurements to recover a (near-)sparse signal. We have shown the utility of Insense in designing DNA probes for the UMD platform.
        </p>
    
        
        [M2] <b>A. Aghazadeh</b>, M. Golbabaee, A. S. Lan, and R. G. Baraniuk, <a href="https://www.sciencedirect.com/science/article/pii/S016516841830121X">Insense: Incoherent sensor selection for sparse signals</a>, <em>Signal Processing</em> 150, 57-65 (2018). <br>
        
        [M3] <b>A. Aghazadeh</b>, M. Golbabaee, A. S. Lan, and R. G. Baraniuk, <a href="https://ieeexplore.ieee.org/abstract/document/8461701">Insense: Incoherent sensor selection for sparse signals</a>, Proc. Intl. Conf. Acoust. Speech Signal Process. (ICASSP), April 2018. <br>  <br>
        
        <a href="https://github.com/amirmohan/Insense">Insense Code on Github</a> <br>
        
    </div>
</div>







<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/building.jpg" alt="Alt Text" width="500">
            <figcaption class="caption">Structural health monitoing of Christopher Wren’s Grade I listed St Mary Abchurch and George Dance’s Mansion House during major tunnelling work taking place under the buildings. Photo taken from the article Cambridge Centre for Smart Infrastructure and Construction.</figcaption>
    </div>

    <div class="toright">
        <p> In a collaboration with folks at Civil Engineering department at Rice University, we developed data-driven tools using sparse signal recovery for the <b>Structural Health Monotiring</b> (SHM) problem.
        </p>
        
        
        [S1] D. Sen, <b>A. Aghazadeh</b>, A. Mousavi, S. Nagarajaiah, and R. G. Baraniuk, <a href="https://www.sciencedirect.com/science/article/abs/pii/S0888327018304898">Sparsity-based data-driven approaches for damage detection in plates</a>, <em>Mech. Syst. Signal Process.</em> 117, 333-346 (2019). <br>
            
        [S2] D. Sen, <b>A. Aghazadeh</b>, A. Mousavi, S. Nagarajaiah, R. G. Baraniuk, and Anand Dabak, <a href="https://www.sciencedirect.com/science/article/abs/pii/S0888327019303784">Data-driven approaches to structural health monitoring of steel pipes</a>, <em>Mech. Syst. Signal Process.</em> 131, 524-537 (2019).
        
        
    </div>
</div>




