---
layout: page
title: Projects
---

<br>
<br>
<br>






<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/gnk_figure.png" alt="Alt Text" width="500">
            <figcaption class="caption">Generalized NK (GNK) model and various neighborhood configurations for proteins.</figcaption>
    </div>


    <div class="toright">
        <p align="justify"> We address a foundational question in biological experiment design: How many experimental measurements do we need to learn a fitness function? We use ideas from random field models in biophysics and compressed sensing in signal processing to address this question. We show that GNK fitness functions with parameters set according to protein structural contacts can be used to approximate the number of samples required to estimate several protein fitness functions.
        </p>
           
                
                [P2] D. Brookes, <b>A. Aghazadeh</b>, and J. Listgarten, <a href="https://www.biorxiv.org/content/10.1101/2021.05.24.445506v1.abstract">On the sparsity of fitness functions and implications for learning</a>, <em> In revision at the Proceedings of the National Academy of Sciences (PNAS)</em>. <br><br>
                

        
    </div>
</div>





<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/en.png" alt="Alt Text" width="500">
            <figcaption class="caption">Regularizing DNNs using our EN method.</figcaption>
    </div>


    <div class="toright">
        <p align="justify"> <b>Epistatic Net (EN)</b> is a combinatorial algorithm to regularize deep neural networks (DNNs) to promote sparsity in their (Walsh-Hadamard) spectral transform. EN significantly improves the accuracy of DNNs in predicting biological function from sequences (e.g., proteins) especially in low-sample regimes. EN subsamples the combinatorial sequence space to induce a sparse-graph-code structure and finds the spectral tranfrom of DNNs using a peeling algorithm in every iteration of a novel ADMM-based regularization scheme.
        </p>
           
                
                [P1] <b>A. Aghazadeh</b>, H. Nisonoff, O. Ocal, Y. Huang, O. Koyluoglu, J. Listgarten, and K. Ramchandran, <a href="https://www.biorxiv.org/content/10.1101/2020.11.24.396994v2">Sparse Epistatic Regularization of Deep Neural Networks for Inferring Fitness Functions</a>, <em> Accepted for publication at Nature Communications</em>. <br><br>
                
        
        <a href="https://github.com/amirmohan/epistatic-net">Code on Github</a> <br>

        
    </div>
</div>






<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/bear.jpg" alt="Alt Text" width="500">
            <figcaption class="caption"> Optimization landscape. Figure adapted from A. Amini et al. “Spatial Uncertainty Sampling for End-to-End Control”. NeurIPS Bayesian Deep Learning 2018.</figcaption>
    </div>


    <div class="toright">
        <p align="justify"> <b>BEAR</b> is a second-order optimization algorithm for ultra-high dimensional feature selection in a sub-linear memory space. BEAR limits the rate of collisions of the stochastic gradients in Count Sketch and significantly improves the memory-accuracy tradeoff in feature selection.
        </p>
           
                
                [S2] <b>A. Aghazadeh</b>, V. Gupta, A. Dewasee, O. Koyluoglu, and K. Ramchandran, <a href="https://msml21.github.io/papers/id26.pdf">BEAR: Sketching BFGS Algorithm for Ultra-High Dimensional Feature Selection in Sublinear Memory</a>, <em>Mathematical and Scientific Machine Learning (MSML)</em>, Aug. 2021. <br> <br>

        
        <a href=" https://github.com/BEAR-algorithm/BEAR">BEAR Code on Github</a> <br>
        <a href="https://drive.google.com/file/d/1BzseCDh8AzGFYo34vR8TUbD0_2YS1vAM/view">My talk at MSML</a> <br>
        <a href="https://msml21.github.io/slides/id26.pdf">My slides at MSML</a> <br>

        
    </div>
</div>







<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/landscape.png" alt="Alt Text" width="500">
            <figcaption class="caption">A rugged fitness landscape. Figure adapted from the <a href="https://en.wikipedia.org/wiki/Sequence_space_(evolution)">Wiki page</a>.</figcaption>
    </div>


    <div class="toright">
        <p align="justify"> <b>CRISPRLand</b> is a spectral platform to explain and analyze the machine learning models trained on the samples from the fitness landscape of DNA repair after CAS9-induced breaks in the Walsh-Hadamard (Fourier) domain. CRISPRLand explains the black-box models in terms of the (higher-order) interactions among the nucleotides in the DNA sequence around the cut site. CRISPRLand infers the entire DNA repair landscape with an exponential speedup compared to conventional methods.
        </p>
           
                
                [C3] <b>A. Aghazadeh</b>, O. Ocal, and K. Ramchandran, <a href="https://www.iscb.org/cms_addon/conferences/ismb2020/tracks/gencompbio">CRISPRLand: Interpretable Large-Scale Inference of DNA Repair Landscape Based on a Spectral Approach</a>, <em> Intelligent Systems for Molecular Biology (ISMB)</em>, July 2020. <br><br>
                
                [C2] <b>A. Aghazadeh</b>, O. Ocal, and K. Ramchandran, <a href="https://academic.oup.com/bioinformatics/article/36/Supplement_1/i560/5870525">CRISPRLand: Interpretable Large-Scale Inference of DNA Repair Landscape Based on a Spectral Approach</a>, <em> Bioinformatics</em>, July 2020. <br><br>
        
        
        <a href="https://github.com/UCBASiCS/CRISPRLand">Code on Github</a> <br>
        <a href="https://youtu.be/IQ9PDs1nqe8">My talk at ISMB</a> <br>

        
    </div>
</div>









<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/crispr.png" alt="Alt Text" width="500">
            <figcaption class="caption">DNA repair after a double-strand break. Figure adapted from the <a href="https://scopeblog.stanford.edu/2019/07/29/crispr-algorithm-predicts-how-well-gene-editing-will-work/">Stanford SCOPE</a> article covering our paper, titled "CRISPR algorithm predicts how well gene editing will work".</figcaption>
    </div>


    <div class="toright">
        <p align="justify"> <b>SPROUT</b> is a machine learning algorithm based on boosted gradient trees to predict the DNA repair outcome after CAS9-induced double-strand breaks in T Cells. Our analysis of DNA repair outcome using SPROUT reveals long insertions (>20 nucleotides) from spatially close locations on the genome.
        </p>
        
                
                [C1] R. Leenay*, <b>A. Aghazadeh</b>*, J. Hiatt*, D. Tse, T. L. Roth, R. Apathy, E. Shifrut, J. F. Hulquist, N. Krogen, Z. Wu, G. Cirolia, H. Canaj, M. Leonetti, A. Marson, A. May, and J. Zou, <a href="https://www.nature.com/articles/s41587-019-0203-2">Large dataset enables prediction of repair after CRISPR–Cas9 editing in primary T cells</a>, <em>Nature Biotechnology</em>, July 2019. <br><br>
        
        
        <a href="https://zou-group.github.io/SPROUT">SPROUT interactive software</a> <br>
        <a href="https://github.com/amirmohan/SPROUT">Code on Github</a> <br>
        <a href="https://figshare.com/projects/Systematic_characterization_of_genome_editing_in_primary_T_cells_reveals_proximal_genomic_insertions_and_enables_machine_learning_prediction_of_CRISPR-Cas9_DNA_repair_outcomes/37166">Data on figshare</a> <br> <br>
        
        News Coverage <br>
        <a href="https://scopeblog.stanford.edu/2019/07/29/crispr-algorithm-predicts-how-well-gene-editing-will-work/">Stanford University</a> <br>
        
    </div>
</div>



<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/sketch.png" alt="Alt Text" width="500">
            <figcaption class="caption">Count Sketch is a data structure from streaming literature. Photo adapted from   <a href="https://blog.acolyer.org/2016/07/21/time-adaptive-sketches-ada-sketches-for-summarizing-data-streams/"><em>the morning paper</em></a> a blog post on time-adaptive sketches for summarizing data streams.</figcaption>
    </div>


    <div class="toright">
        <p align="justify"> <b>MISSION</b> is a first-order optimization algorithm for sublinear-memory feature selection in massively high-dimensional datasets. The key idea behind the algorithm is to store and iteratively update the model parameters in an efficient data structure from streaming called Count Sketch.
        </p>
        
                
        [S1] <b>A. Aghazadeh</b>*, R. Spring*, D. LeJeune, G. Dasarathy, A. Shrivastava and R. G. Baraniuk, <a href="https://icml.cc/Conferences/2018/ScheduleMultitrack?event=2916">MISSION: Ultra Large-Scale Feature Selection using Count Sketches</a>, <em>International Conference on Machine Learning (ICML)</em>, July 2018. <br>  <br>
        
    
        
        <a href="https://github.com/rdspring1/MISSION">Code on Github</a> <br>

        
    </div>
</div>




<hr>



<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="canter" src="../assets/images/insense.pdf" alt="Alt Text" width="500">
            <figcaption class="caption">Universal Microbial Diagnostic (UMD) platform shows the steps to identify the composition of a microbial sample using randomized DNA probes. Photo is adapted from our <em>Science Advances</em> paper [M1].</figcaption>
    </div>

    <div class="toright">
        <p align="justify"> <b>Insense</b> is an optimization algorithm to select optimal linear measurements to recover a (near-)sparse signal. We have shown the utility of Insense in designing DNA probes for the UMD platform.
        </p>
    
        
        [M3] <b>A. Aghazadeh</b>, M. Golbabaee, A. S. Lan, and R. G. Baraniuk, <a href="https://www.sciencedirect.com/science/article/pii/S016516841830121X">Insense: Incoherent sensor selection for sparse signals</a>, <em>Signal Processing</em> 150, 57-65 (2018). <br> <br>
        
        [M2] <b>A. Aghazadeh</b>, M. Golbabaee, A. S. Lan, and R. G. Baraniuk, <a href="https://ieeexplore.ieee.org/abstract/document/8461701">Insense: Incoherent sensor selection for sparse signals</a>, <em>International Conference on Acoustics, Speech, and Signal Processing (ICASSP)</em>, April 2018. <br>  <br>
        
        <a href="https://github.com/amirmohan/Insense">Code on Github</a> <br> <br>
        
        News Coverage <br>
        <a href="https://nuit-blanche.blogspot.com/2017/03/emmanuel-candes-in-paris-insense.html">Nuit-Blanche Blog Post</a> <br><br>
        
        
    </div>
</div>








<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/infinity.pdf" alt="Alt Text" width="500">
            <figcaption class="caption">Robust hashing via minimizing the worst-case distortion results in a better set of nearest neighbors.</figcaption>
    </div>


    <div class="toright">
        <p align="justify"> <b>RHash</b> is a fast algorithm to learn <b>robust data-dependent hash functions</b>. It minimizes the worst-case L-infinity norm distortion instead of the more conventional average distortion to learn hash functions.

        </p>
        
        
        [H1] <b>A. Aghazadeh</b>, A. S. Lan, A. Shrivastava and R. G. Baraniuk, <a href="https://www.ijcai.org/Proceedings/2017/0192.pdf">RHash: Robust Hashing via L∞-norm Distortion</a>, <em>International Joint Conference on Artificial Intelligence (IJCAI)</em>, Aug. 2017. <br>
        
        
    </div>
</div>






<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/UMD.pdf" alt="Alt Text" width="500">
            <figcaption class="caption">Universal Microbial Diagnostic (UMD) platform shows the steps to identify the composition of a microbial sample using randomized DNA probes. Photo is adapted from our <em>Science Advances</em> paper [M1].</figcaption>
    </div>

    <div class="toright">
        <p align="justify"> <b>Universal Microbial Diagnostics</b> (UMD) is a new sensing paradigm that we have developed at Rice University to detect microbial agents (e.g., superbugs and viruses) using a small number of target-agnostic randomized DNA probes. The inference algorithm solves a compressive sensing problem to identify the composition of the microbial samples.
        </p>
    
        
        [M1] <b>A. Aghazadeh</b>*, A. Y. Lin*, M. A. Sheikh*, R. A. Drezek, R. G. Baraniuk, et al., <a href="https://advances.sciencemag.org/content/2/9/e1600025?utm_source=TrendMD&utm_medium=cpc&utm_campaign=TrendMD_1">Universal microbial diagnostics using random DNA probes</a>, <em>Science Advances</em> 2, e1600025 (2016). <br> <br>
        
        
        
        <a href="https://patents.google.com/patent/US20180355411A1/en">US Patent</a> <br>
        <a href="https://www.ece.rice.edu/news/21704-REA-Awards">Hershel M. Rich Invention Award</a> <br><br>
    
        
        News Coverage <br>
        <a href="http://www.houstonchronicle.com/local/prognosis/article/New-probe-designed-by-Rice-and-Baylor-team-could-9368593.php">Houston Chronicle</a> <br>
        
        <a href="https://www.genomeweb.com/molecular-diagnostics/universal-microbial-diagnostics-approach-using-random-dna-probes-promises#.XqXD3y-z10s">GenomeWeb</a> <br>
        
        <a href="https://www.pharmaceutical-journal.com/news-and-analysis/research-briefing/scientists-create-universal-microbial-screening-method/20201883.article">The Pharmaceutical Journal</a> <br>

        <a href="https://www.biocentury.com/article/279557/the-first-catch-all-diagnostic-for-infections">BioCentury Innovations</a> <br>
        
        
        <a href="https://www.labroots.com/trending/microbiology/4207/technology-easily-identifies-bacterial-pathogens">Labroots</a> <br>
        
        
    </div>
</div>



<hr>

<div>
<div class="side-by-side">
    <div class="toleft">
        <img class="image" src="../assets/images/building.jpg" alt="Alt Text" width="500">
            <figcaption class="caption">Structural health monitoring of Christopher Wren’s Grade I listed St Mary Abchurch and George Dance’s Mansion House during major tunnelling work taking place under the buildings. Photo taken from the article Cambridge Centre for Smart Infrastructure and Construction.</figcaption>
    </div>

    <div class="toright">
        <p align="justify"> In a collaboration with folks at the Civil Engineering department at Rice University, we developed data-driven tools using sparse signal recovery for the <b>Structural Health Monitoring</b> (SHM) problem.
        </p>
        
        
        [S2] D. Sen, <b>A. Aghazadeh</b>, A. Mousavi, S. Nagarajaiah, and R. G. Baraniuk, <a href="https://www.sciencedirect.com/science/article/abs/pii/S0888327018304898">Sparsity-based data-driven approaches for damage detection in plates</a>, <em>Mechanical System and Signal Processing (MSSP)</em> 117, 333-346 (2019). <br> <br>
            
        [S1] D. Sen, <b>A. Aghazadeh</b>, A. Mousavi, S. Nagarajaiah, R. G. Baraniuk, and A. Dabak, <a href="https://www.sciencedirect.com/science/article/abs/pii/S0888327019303784">Data-driven approaches to structural health monitoring of steel pipes</a>, <em>Mechanical System and Signal Processing (MSSP)</em> 131, 524-537 (2019).
        
        
    </div>
</div>




